### DISCUSSION TITLE
SupportBrowserHistory.php rewrites component's public variables

### DISCUSSION THREAD

**Discussion by almazdanilov** [livewire/livewire#3978] [7. Bugs (archived)] [answered]
Livewire v2.6.7
Laravel v8.62.0

Should public variables of component be rewritten by query string parameters?
As I understood the documentation of Livewire on https://laravel-livewire.com/docs/2.x/query-string
to achieve such abillity, we have to add to livewire component:
`protected $queryString = ['queryParamName'];`
But Features/SupportBrowserHistory.php rewrites component's public variable, if same parameter name written in query.

**/catalog?vendor=1**
```
class Catalog extends Component
{
    public $vendor = '';
    public function mount() {
        dd($vendor);
    }
}
```
This code returns 1.

SupportBrowserHistory.php, 39'th line of code does this.
`$component->$property = $decoded === null ? $fromQueryString : $decoded;`

Maybe I misunderstood something, and this behaviour is normal?

**Comment by joshhanley** [comment]
@almazdanilov I've just tried to replicate your issue, but I can't replicate it and it all seems to be working properly to me.

Are you able to share your component code and route definition so we can replicate the issue?

See below for the test I did.

Thanks!

![image](https://user-images.githubusercontent.com/882837/136627311-4b8be4b9-1f26-42db-b1bc-ccaeaca6a15e.png)

**Comment by almazdanilov** [comment, marked as answer]
Sorry. My fault. My component is too complicated and I just saw that I've used that variable on $queryString array. Need a day off.
Please delete this discussion.

**Reply by joshhanley** [reply]
All good! Give yourself a break 🙂
